server/bundles/io.cloudbeaver.service.ldap.auth/src/io/cloudbeaver/service/ldap/auth/LdapAuthProvider.java

```diff
@@ -323,19 +350,24 @@ private Map<String, Object> authenticateLdap(
         String password,
         LdapSettings ldapSettings,
         @Nullable String login,
-        Hashtable<String, String> environment
+        Hashtable<String, String> environment,
+        Map<String, Object> userData
     ) throws DBException {
         environment.put(Context.SECURITY_PRINCIPAL, userDN);
         environment.put(Context.SECURITY_CREDENTIALS, password);
         DirContext userContext = null;
         try {
             userContext = new InitialDirContext(environment);
-            Map<String, Object> userData = new HashMap<>();
-            userData.put(LdapConstants.CRED_USERNAME, findUserNameFromDN(userDN, ldapSettings));
-            userData.put(LdapConstants.CRED_FULL_DN, userDN);
-            userData.put(LdapConstants.CRED_SESSION_ID, UUID.randomUUID());
-            if (login != null) {
-                userData.put(LdapConstants.CRED_DISPLAY_NAME, login);
+            if (CommonUtils.isNotEmpty(login)) {
+                userData.put(LdapConstants.CRED_USERNAME, login);
+                userData.put(LdapConstants.CRED_FULL_DN, userDN);
+                userData.put(LdapConstants.CRED_SESSION_ID, UUID.randomUUID());
+                userData.put(LdapConstants.CRED_DISPLAY_NAME, findUserNameFromDN(userDN, ldapSettings));
```

Comment:
userData.put(LdapConstants.CRED_SESSION_ID, UUID.randomUUID());
userData.put(LdapConstants.CRED_DISPLAY_NAME, findUserNameFromDN(userDN, ldapSettings));

move duplicated lines out of if scope